mcstatus/__main__.py

```diff
@@ -1,75 +1,152 @@
 from __future__ import annotations
 
+import dns.resolver
+import sys
+import json
 import argparse
 import socket
-from json import dumps as json_dumps
+import warnings
+import dataclasses
+from typing import TYPE_CHECKING
 
-from mcstatus import JavaServer
+from mcstatus import JavaServer, BedrockServer
+from mcstatus.responses import JavaStatusResponse
+from mcstatus.motd import Motd
 
+if TYPE_CHECKING:
+    SupportedServers = JavaServer | BedrockServer
 
-def ping(server: JavaServer) -> None:
-    print(f"{server.ping()}ms")
 
+def _motd(motd: Motd) -> str:
+    """Formats MOTD for human-readable output, with leading line break
+    if multiline."""
+    s = motd.to_ansi()
+    return f"\n{s}" if "\n" in s else f" {s}"
 
-def status(server: JavaServer) -> None:
+
+def _kind(serv: SupportedServers) -> str:
+    if isinstance(serv, JavaServer):
+        return "Java"
+    elif isinstance(serv, BedrockServer):
+        return "Bedrock"
+    else:
+        raise ValueError(f"unsupported server for kind: {serv}")
+
+
+def _ping_with_fallback(server: SupportedServers) -> float:
+    # bedrock doesn't have ping method
+    if isinstance(server, BedrockServer):
+        return server.status().latency
+
+    # try faster ping packet first, falling back to status with a warning.
+    ping_exc = None
+    try:
+        return server.ping(tries=1)
+    except Exception as e:
+        ping_exc = e
+
+    latency = server.status().latency
+
+    address = f"{server.address.host}:{server.address.port}"
+    warnings.warn(
```

Comment:
I wonder if we should just print this to stderr instead of using warnings here, `warnings` module is great for libraries, where people use the API to call methods. But CLI users shouldn't need to be familiar with where in the code that warning was produced nor about the other metadata that warnings module adds here, it only makes the message less readable IMO.

Reply:
printing now